crates/bcr-ebill-api/src/service/notification_service/nostr.rs

```diff
@@ -188,59 +213,88 @@ impl NostrConsumer {
+
+                // subscribe only to private messages sent to our pubkey
+                current_client
+                    .subscribe(
+                        Filter::new()
+                            .pubkey(public_key)
+                            .kind(Kind::GiftWrap)
+                            .since(offset_ts),
+                    )
+                    .await
+                    .expect("Failed to subscribe to Nostr events");
 
-        client
-                .client
-                .handle_notifications(|note| async {
-                    if let Some((envelope, sender, event_id, time)) =
-                        client.unwrap_envelope(note).await
-                    {
-                        if !offset_store.is_processed(&event_id.to_hex()).await? {
-                            if let Ok(sender_npub) = sender.to_bech32() {
-                                let sender_node_id = sender.to_hex();
-                                trace!("Received event: {envelope:?} from {sender_npub:?} (hex: {sender_node_id})");
-                                // We use hex here, so we can compare it with our node_ids
-                                // TODO: re-enable after presentation: if contact_service.is_known_npub(&sender_node_id).await? {
-                                    trace!("Processing event: {envelope:?}");
-                                    handle_event(envelope, &node_id, &event_handlers).await?;
-                                // }
-                            }
-
-                            // store the new event offset
-                            add_offset(&offset_store, event_id, time, true).await;
+                let inner = current_client.clone();
+                current_client
+                    .client
+                    .handle_notifications(move |note| {
+                        let client = inner.clone();
+                        let event_handlers = event_handlers.clone();
+                        let offset_store = offset_store.clone();
+                        let node_id = node_id.clone();
+                        let client_id = client_id.clone();
+
+                        async move {
+                            if let Some((envelope, sender, event_id, time)) =
+                                client.unwrap_envelope(note).await
+                            {
+                                if !offset_store.is_processed(&event_id.to_hex()).await? {
+                                    let sender_npub = sender.to_bech32();
+                                    let sender_node_id = sender.to_hex();
+                                    trace!("Received event: {envelope:?} from {sender_npub:?} (hex: {sender_node_id}) on client {client_id}");
+                                    // We use hex here, so we can compare it with our node_ids
+                                    // TODO: re-enable after presentation: if contact_service.is_known_npub(&sender_node_id).await? {
```

Comment:
is this todo still relevant?

Reply:
Yes, it need the same treatment than the one I added for the consumer instances (so whitelist all our own identities). After that we can re-activate this.